src/ClientResolver.php

```diff
@@ -1130,12 +1140,14 @@ public static function _default_auth_scheme_resolver(array $args)
     }
 
     public static function _apply_auth_scheme_preference(
-        string|array|null $value,
+        string|array|null &$value,
         array &$args
     ): void
     {
         // Not provided user's preference auth scheme list
         if (empty($value)) {
+            $value = null;
+            $args['config']['auth_scheme_preference'] = $value;
```

Comment:
Why is needed to set this value to null in this config. Also, Is it guaranteed that `auth_scheme_preference` will be a valid property in `$args['config']`?, otherwise this will cause a runtime error.

Reply:
Currently, when no value is found in the ini file or env, it enters this function as an empty string.  Passing by reference and changing `$value` to `null` here changes both `$args['auth_scheme_preference']` and (as shown) sets `$args['config']['auth_scheme_preference']` to null.  `$args['config']` is just another arbitrary array, so setting to `null` shouldn't cause any errors.